From 8a7773443c2ea0e0afa93d366edab9445b5d63b8 Mon Sep 17 00:00:00 2001
From: johnzhax <johnx.zhang@intel.com>
Date: Wed, 29 Aug 2018 15:56:34 +0800
Subject: [PATCH] Fix the issue unable to send outgoing call in Contact app

Ignore the judgement for the configuration of config_voice_capable in
automotive device while sending an outgoing call even it is false.

Change-Id: Ic5dee2f2e011e34f2e17ca01247225c46c80c8be
Tracked-On: OAM-75510
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 .../contacts/compat/TelephonyManagerCompat.java        | 10 +++++++++-
 .../preference/DisplayOptionsPreferenceFragment.java   |  4 +---
 .../android/contacts/util/PhoneCapabilityTester.java   |  4 +---
 3 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/com/android/contacts/compat/TelephonyManagerCompat.java b/src/com/android/contacts/compat/TelephonyManagerCompat.java
index 07523b1..965fa68 100644
--- a/src/com/android/contacts/compat/TelephonyManagerCompat.java
+++ b/src/com/android/contacts/compat/TelephonyManagerCompat.java
@@ -17,6 +17,9 @@
 package com.android.contacts.compat;
 
 import android.net.Uri;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.telecom.PhoneAccountHandle;
 import android.telephony.TelephonyManager;
@@ -39,10 +42,15 @@ public class TelephonyManagerCompat {
      * on any device with a telephony radio, even if the device is
      * data-only.
      */
-    public static boolean isVoiceCapable(@Nullable TelephonyManager telephonyManager) {
+    public static boolean isVoiceCapable(@NonNull Context context) {
+        TelephonyManager telephonyManager =
+                (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
         if (telephonyManager == null) {
             return false;
         }
+        if (context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE)) {
+            return true;
+        }
         if (CompatUtils.isLollipopMr1Compatible()
                 || CompatUtils.isMethodAvailable(TELEPHONY_MANAGER_CLASS, "isVoiceCapable")) {
             // isVoiceCapable was unhidden in L-MR1
diff --git a/src/com/android/contacts/preference/DisplayOptionsPreferenceFragment.java b/src/com/android/contacts/preference/DisplayOptionsPreferenceFragment.java
index 82e90b3..845be94 100644
--- a/src/com/android/contacts/preference/DisplayOptionsPreferenceFragment.java
+++ b/src/com/android/contacts/preference/DisplayOptionsPreferenceFragment.java
@@ -38,7 +38,6 @@ import android.provider.ContactsContract.Profile;
 import android.support.design.widget.Snackbar;
 import android.support.v4.content.LocalBroadcastManager;
 import android.telecom.TelecomManager;
-import android.telephony.TelephonyManager;
 import android.text.BidiFormatter;
 import android.text.TextDirectionHeuristics;
 import android.view.LayoutInflater;
@@ -294,8 +293,7 @@ public class DisplayOptionsPreferenceFragment extends PreferenceFragment
             getPreferenceScreen().removePreference(findPreference(KEY_DISPLAY_ORDER));
         }
 
-        final boolean isPhone = TelephonyManagerCompat.isVoiceCapable(
-                (TelephonyManager) getContext().getSystemService(Context.TELEPHONY_SERVICE));
+        final boolean isPhone = TelephonyManagerCompat.isVoiceCapable(getContext());
         final boolean showBlockedNumbers = isPhone && ContactsUtils.FLAG_N_FEATURE
                 && BlockedNumberContract.canCurrentUserBlockNumbers(getContext());
         if (!showBlockedNumbers) {
diff --git a/src/com/android/contacts/util/PhoneCapabilityTester.java b/src/com/android/contacts/util/PhoneCapabilityTester.java
index 9753b88..5df9aa3 100644
--- a/src/com/android/contacts/util/PhoneCapabilityTester.java
+++ b/src/com/android/contacts/util/PhoneCapabilityTester.java
@@ -25,7 +25,6 @@ import android.net.Uri;
 import android.net.sip.SipManager;
 import android.provider.MediaStore;
 import android.provider.Telephony;
-import android.telephony.TelephonyManager;
 
 import com.android.contacts.ContactsUtils;
 import com.android.contacts.compat.TelephonyManagerCompat;
@@ -62,8 +61,7 @@ public final class PhoneCapabilityTester {
     }
 
     private static void initialize(Context context) {
-        sIsPhone = TelephonyManagerCompat.isVoiceCapable(
-                (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE));
+        sIsPhone = TelephonyManagerCompat.isVoiceCapable(context);
         sIsSipPhone = sIsPhone && SipManager.isVoipSupported(context);
         sIsInitialized = true;
     }
-- 
2.17.1

